src/app/nutrition/dialogs/food-log-rows.tsx

```diff
@@ -245,3 +264,95 @@ export function TotalsRow({
 
   return <MealTypeRows summary={summary} />;
 }
+
+export function NutritionGoalsRow({ totals }: { totals: FoodLogSummary }) {
+  const calories = useAtomValue(caloriesGoalAtom);
+  const carbs = useAtomValue(carbsGoalAtom);
+  const fat = useAtomValue(fatGoalAtom);
+  const fiber = useAtomValue(fiberGoalAtom);
+  const protein = useAtomValue(proteinGoalAtom);
+  const sodium = useAtomValue(sodiumGoalAtom);
+  const useNetCars = useAtomValue(enableNetCarbsAtom);
+  const caloriesPercentage = parseFloat((totals.calories * 100/ calories).toFixed(2));
+  const totalsCarbs = useNetCars ? totals.carbs - totals.fiber : totals.carbs;
+  const summary: MealTypeSummary = {
+    id: MealType.Anytime,
+    name: "Remaining",
+    foods: [
```

Comment:
This should just render table cells directly instead of creating fake foods and rendering it using the MealTypeRows component. Then we don't need special logic for `readonly` and other weirdness from pretending that the goal has to fit into food type.

For example, the Nutrition Goals % row should show % after each of the percentages so it's clear that it's not in g/mg like the other values, and trying to do that with the food row rendering would require even more custom logic.

Reply:
I was trying to not duplicate code but you're right, it was pretty dirty :sweat_smile: